Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -143,9 +146,18 @@ namespace AzToolsFramework
                 }
 
                 // Create the Prefab
-                instanceToCreate = prefabEditorEntityOwnershipInterface->CreatePrefab(
-                    entities, AZStd::move(instancePtrs), m_prefabLoaderInterface->GenerateRelativePath(absolutePath),
-                    commonRootEntityOwningInstance);
+                if (saveToDisk)
+                {
+                    instanceToCreate = prefabEditorEntityOwnershipInterface->CreatePrefab(
+                        entities, AZStd::move(instancePtrs), m_prefabLoaderInterface->GenerateRelativePath(filePath),
+                        commonRootEntityOwningInstance);
+                }
+                else
+                {
+                    instanceToCreate = prefabEditorEntityOwnershipInterface->CreatePrefab(
+                        entities, AZStd::move(instancePtrs), filePath,
+                        commonRootEntityOwningInstance);
```

Comment:
We should always try to store relative paths in the prefabs to keep behavior consistent if a user ever chooses to save the prefab to disk later after creating it without saving first.

Reply:
Fixed.